python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py

```diff
@@ -193,9 +220,11 @@ def _get_result(row):
                 )
                 value = apply_transforms_fn(value, field=field, repeated=is_repeated)
                 if is_repeated:
-                    value = [
-                        _cast_value(self.node.ctx, v, field.data_type) for v in value
-                    ]
+                    value = (
+                        [_cast_value(self.node.ctx, v, field.data_type) for v in value]
+                        if not _is_na(value)
+                        else value
```

Comment:
Would this function benefit an early return as well?

Reply:
Not sure I understand (we return result, not value) but I changed it to a if..elif..else, it should be more readable now :)